src/go/configgenerator/filterconfig/filter_generator.go

```diff
@@ -266,6 +275,32 @@ func updateProtoDescriptor(service *confpb.Service, apiNames []string, descripto
 	return newData, nil
 }
 
+func preserveDefaultHttpBinding(httpRule *ahpb.HttpRule, defaultPath string) {
+	defaultBinding := &ahpb.HttpRule{Pattern: &ahpb.HttpRule_Post{defaultPath}, Body: "*"}
+
+	// Check existence of the default binding in httpRule's additional_bindings to avoid duplication.
+	defaultBindingExisted := false
+	for _, addtionalBinding := range httpRule.AdditionalBindings {
+		if proto.Equal(addtionalBinding, defaultBinding) {
+			defaultBindingExisted = true
+			break
+		}
+	}
+	// check if httpRule is the same as default binding, ignore the difference in fields selector and
+	// additional_bindings.
+	defaultBinding.Selector = httpRule.GetSelector()
```

Comment:
How do you guarantee that original http rule has a selector?
Similarly, how do you know the additional bindings above had no selector?

My concern is that this will break in the future. What if we start adding selector to additional binding? Then the loop above will never work

Reply:
From what I know, if the original http rule doesn't have a selector, an error will be thrown before the execution of this part of the code.
There's no guarantee about the selectors, that's why the difference of selectors are ignored. If we start adding selector to additional binding, then the default binding will always be added.